src/game/savegame/savegame_bson.c

```diff
@@ -1106,6 +1137,22 @@ static struct json_object_s *Savegame_BSON_DumpLara(LARA_INFO *lara)
     return lara_obj;
 }
 
+static struct json_object_s *SaveGame_BSON_DumpCurrentMusic(void)
+{
+    struct json_object_s *current_music_obj = json_object_new();
+    json_object_append_int(
+        current_music_obj, "current_track", Music_CurrentTrack());
+    int64_t timestamp = 0;
+    if (Music_CurrentTrack()) {
+        timestamp = Music_GetTimestamp(Music_CurrentTrack());
+    }
+    int32_t *timestamp_arr = (int32_t *)&timestamp;
+    json_object_append_int(current_music_obj, "timestamp1", timestamp_arr[0]);
+    json_object_append_int(current_music_obj, "timestamp2", timestamp_arr[1]);
```

Comment:
I only noticed this now. Hah, clever. Could we make a `json_object_append_int64` though for this? JSON is plenty capable of holding arbitrary-precision numbers and the limitation of our code shouldn't be reflected in the files IMO. @lahm86 what's your opinion on this?

Reply:
Ok I can do that. I didn't want to rely on a generic `int` since the timestamp is 64 bit.

Reply:
I like the array approach too, but agree, having JSON do the work sounds great.